tests/baselines/reference/renamingDestructuredPropertyInFunctionType.js

```diff
@@ -101,14 +101,14 @@ type O = {
     c: number;
 };
 type F1 = (arg: number) => any;
-type F2 = ({ a }: O) => any;
-type F3 = ({ a, b, c }: O) => any;
-type F4 = ({ a }: O) => any;
-type F5 = ({ a, b, c }: O) => any;
+type F2 = ({ a: string }: O) => any;

```

Comment:
> The error for unused renames is still there. I did not remove it. It only seems to pop up on some kinds of declarations for example function types `let o: ({n : a}: any) => void;`

Just to point out an example from the PR, this is code that I believed would have gained an error: https://www.typescriptlang.org/play?#code/C4TwDgpgBA8lC8UDeBYAUFTUCGB+AXFAM7ABOAlgHYDmA3OllAEaGUCuAtkxKfRlgGNWnbr3QBfPulCQoAMQBMCKAAokOQiQo0o4wjACUCAHw5KIWkA

Yet, there's no error baseline. That being said, it looks like for declaration files we straight up skip checking the final list:

```
if (!node.isDeclarationFile) {
    checkPotentialUncheckedRenamedBindingElementsInTypes();
}
```

So, I think everything is fine? (Other than that I don't think it makes any sense that we still collect `potentialUnusedRenamedBindingElementsInTypes` in declaration files only to not read them later, which is not new.)